lib/matplotlib/colors.py

```diff
@@ -56,6 +58,12 @@
 from matplotlib import _api, _cm, cbook, scale
 from ._color_data import BASE_COLORS, TABLEAU_COLORS, CSS4_COLORS, XKCD_COLORS
 
+Color = Union[
```

Comment:
I can certainly see an argument in either direction here... I initially started with a `_typing.pyi` file that I filled with a handful of type definitions that were used in multiple files.

When I was done I somewhat felt that the definitions that were there (~6 of them, highlighted in the third bullet point above) were narrowly scoped enough that placing them together instead of with their use cases was a little awkward, but I could certainly see doing a `typing.py` instead of where I ended up.

I do think that it should be public (and available at runtime so that downstream can type check without necessarily doing `if TYPE_CHECKING:` blocks) rather than what I had initially done. But it can be more explicit about being typing.

the `_typing.pyi` file has been wiped from the git history since it was both added and deleted and was triggering the Cleanliness check to fail.

Even if we don't end up moving to a `typing` module, I was considering whether `Color` in particular should be `ColorType` or similar. Especially given conversation in #24859 about potentially creating an actual `Color` class to allow some level of comparisons between string forms and RGB(A) forms of colors.

Reply:
I am :+1: on calling this `ColorType` (`ColourType`)?